os/macosx/openttd.entitlements

```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>com.apple.security.cs.allow-dyld-environment-variables</key>
+	<true/>
+	<key>com.apple.security.cs.disable-library-validation</key>
```

Comment:
We already add some entitlements, but in a different place:

https://github.com/OpenTTD/OpenTTD/blob/master/.github/workflows/release-macos.yml#L133

Possibly a simpler solution would be to add `-o library` there. Or at least synchronize the two, so we don't have two places doing somewhat the same work.

Reply:
We need both for proper Steam support, and the DYLD environment flags entitlement isn't supported by options as far as I know.